pkg/operator/ceph/cluster/mon/health.go

```diff
@@ -357,21 +358,36 @@ func (c *Cluster) addOrRemoveExternalMonitor(status client.MonStatusResponse) (b
 				monPort := cephutil.GetPortFromEndpoint(endpoint)
 				logger.Infof("new external mon %s found: %s, adding it", mon.Name, endpoint)
 				c.ClusterInfo.Monitors[mon.Name] = cephconfig.NewMonInfo(mon.Name, monIP, monPort)
-				changed = true
+			} else {
+				logger.Debugf("mon %s is not in quorum and not in ClusterInfo", mon.Name)
 			}
-			logger.Debugf("mon %s is not in quorum and not in ClusterInfo", mon.Name)
+			changed = true
 		} else {
 			// mon is in ClusterInfo
 			logger.Debugf("mon %s is in ClusterInfo, let's test if it's in quorum", mon.Name)
 			if !inQuorum {
-				// this mon was in clusterInfo but not part of the quorum anymore
-				// let's remove it from ClusterInfo
+				// this mon was in clusterInfo but is not part of the quorum anymore
+				// thus don't add it again to ClusterInfo
 				logger.Infof("monitor %s is not part of the external cluster monitor quorum, removing it", mon.Name)
-				delete(c.ClusterInfo.Monitors, mon.Name)
+				changed = true
+			} else {
+				// this mon was in clusterInfo and is still in the quorum
+				// add it again
+				c.ClusterInfo.Monitors[mon.Name] = oldClusterInfoMonitors[mon.Name]
+				logger.Debugf("everything is fine mon %s in the clusterInfo and its quorum status is %v", mon.Name, inQuorum)
+			}
+		}
+	}
+	// compare old clusterInfo with new ClusterInfo
+	if len(oldClusterInfoMonitors) != len(c.ClusterInfo.Monitors) {
+		changed = true
+	} else {
+		// check for similarity
```

Comment:
Why is this necessary?
If we know it's true then let's just return true.

Reply:
Sorry, but that is exactly the problem. If the new ClusterInfo ist the same as the old one, false should be returned.
All I do is a set compare, if the set differs I return true otherwise false.

Reply:
Sorry I got confused, the first if just checks the number if mon and the `else` checks if they are actually the same. Looks good!